lib/src/compiler/linters.rs

```diff
@@ -114,6 +115,145 @@ impl LinterInternal for RuleName {
+            error: false,
+        };
+        Ok(tags)
+    }
+
+    /// Specifies whether the linter should produce an error instead of a
+    /// warning.
+    ///
+    /// By default, the linter raises warnings about rule names that don't match
+    /// the regular expression. This setting allows turning such warnings into
+    /// errors.
+    pub fn error(mut self, yes: bool) -> Self {
+        self.error = yes;
+        self
+    }
+}
+
+impl LinterInternal for Tags {
+    fn check(
+        &self,
+        report_builder: &ReportBuilder,
+        rule: &ast::Rule,
+    ) -> LinterResult {
+        if rule.tags.is_none() {
+            return LinterResult::Ok;
+        }
+
+        let mut results: Vec<Warning> = Vec::new();
+        let tags = rule.tags.as_ref().unwrap();
+        if !self.allow_list.is_empty() {
+            for tag in tags.iter() {
+                if !self.allow_list.contains(&tag.name.to_string()) {
+                    if self.error {
+                        return LinterResult::Err(errors::UnknownTag::build(
+                            report_builder,
+                            tag.span().into(),
+                            tag.name.to_string(),
+                            Some(format!(
+                                "Allowed tags: {}",
+                                self.allow_list.join(", ")
+                            )),
+                        ));
+                    } else {
+                        results.push(warnings::UnknownTag::build(
+                            report_builder,
+                            tag.span().into(),
+                            tag.name.to_string(),
+                            Some(format!(
+                                "Allowed tags: {}",
+                                self.allow_list.join(", ")
+                            )),
+                        ));
+                    }
+                }
+            }
+        } else {
+            let compiled_regex = self.compiled_regex.as_ref().unwrap();
+
+            for tag in tags.iter() {
+                if !compiled_regex.is_match(tag.name) {
+                    if self.error {
+                        return LinterResult::Err(errors::InvalidTag::build(
+                            report_builder,
+                            tag.span().into(),
+                            tag.name.to_string(),
+                            self.regex.clone(),
+                        ));
+                    } else {
+                        results.push(warnings::InvalidTag::build(
+                            report_builder,
+                            tag.span().into(),
+                            tag.name.to_string(),
+                            self.regex.clone(),
+                        ));
+                    }
+                }
+            }
+        }
+
+        return if results.is_empty() {
+            LinterResult::Ok
+        } else {
+            return LinterResult::Warns(results);
```

Comment:
The `return` keyword can be removed here, you are already returning the result from th `if` statement.